crapssim/bet.py

```diff
@@ -670,6 +765,88 @@ def __repr__(self) -> str:
         return f"Place({self.winning_numbers[0]}, amount={self.amount})"
 
 
+class Buy(_SimpleBet):
```

Comment:
The tricky part with Buy and Lay bets is that if you do a $20 bet (with $1 vig), and the table settings are such that you pay the vig on each bet, then the bankroll should decrease by $21 for a $20 bet. The current logic doesn't do this, which can be seen in the run_examples.py if running verbosely. It matters because with $100 you can't make 5 buy bets for $20, you can only make 4, for example. And thus the losses are not really accurate either (it's a $21 loss for such a bet, not $20).

I don't have a quick fix here unfortunately. Maybe this bet needs two different "amounts" for proper bookkeeping